### PR TITLE
One name for MulticlassClassification

### LINKED ISSUE #2623: Multiclass in API has many names
When we refer to multiclass in the API, it has many names: `MulticlassClassification`, `MultiClassClassifierMetrics` are just two examples. For the V1 API, does it make sense to stick to one naming scheme?

### PR DESCRIPTION
**Pull request by artidoro**
Fixes #2623.

In this PR I make sure that the case of `Multiclass` is consistent and rename:
- `MultiClass[...]` to `Multiclass[...]`

With PR #2903 this solves #2623.

### CODE REVIEW

File: src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ private static class DefaultEvaluatorTable
                         var tmp = new Dictionary<string, Func<IHostEnvironment, IMamlEvaluator>>
                         {
                             { AnnotationUtils.Const.ScoreColumnKind.BinaryClassification, env => new BinaryClassifierMamlEvaluator(env, new BinaryClassifierMamlEvaluator.Arguments()) },
-                            { AnnotationUtils.Const.ScoreColumnKind.MultiClassClassification, env => new MultiClassMamlEvaluator(env, new MultiClassMamlEvaluator.Arguments()) },
+                            { AnnotationUtils.Const.ScoreColumnKind.MultiClassClassification, env => new MultiClassClassificationMamlEvaluator(env, new MultiClassClassificationMamlEvaluator.Arguments()) },
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs]: >MultiClassClassificationMamlEvaluator [](start = 105, length = 37)

Multiclass - note the casing

File: src/Microsoft.ML.Data/TrainCatalog.cs [hunk 2]
```diff
@@ -519,13 +519,13 @@ internal MulticlassClassificationTrainers(MulticlassClassificationCatalog catalo
         /// <param name="seed">Seed for the random number generator used to select rows for cross-validation folds.</param>
         /// <returns>Per-fold results: metrics, models, scored datasets.</returns>
         /// <returns>Per-fold results: metrics, models, scored datasets.</returns>
-        public CrossValidationResult<MultiClassClassifierMetrics>[] CrossValidate(
+        public CrossValidationResult<MultiClassClassificationMetrics>[] CrossValidate(
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/TrainCatalog.cs]: >MultiClassClassificationMetrics [](start = 37, length = 31)

Multiclass not MultiClass

File: src/Microsoft.ML.Data/EntryPoints/InputBuilder.cs [hunk 3]
```diff
@@ -839,11 +839,11 @@ public static class PipelineSweeperSupportedMetrics
         {
             public static new string ToString() => "SupportedMetric";
             public const string Auc = BinaryClassifierEvaluator.Auc;
-            public const string AccuracyMicro = Data.MultiClassClassifierEvaluator.AccuracyMicro;
-            public const string AccuracyMacro = MultiClassClassifierEvaluator.AccuracyMacro;
+            public const string AccuracyMicro = Data.MultiClassClassificationEvaluator.AccuracyMicro;
+            public const string AccuracyMacro = MultiClassClassificationEvaluator.AccuracyMacro;
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/EntryPoints/InputBuilder.cs]: >MultiClassClassificationEvaluator [](start = 48, length = 33)

Multiclass

File: src/Microsoft.ML.Data/TrainCatalog.cs [hunk 4]
```diff
@@ -485,22 +485,22 @@ internal MulticlassClassificationTrainers(MulticlassClassificationCatalog catalo
         /// <param name="label">The name of the label column in <paramref name="data"/>.</param>
         /// <param name="score">The name of the score column in <paramref name="data"/>.</param>
         /// <param name="predictedLabel">The name of the predicted label column in <paramref name="data"/>.</param>
-        /// <param name="topK">If given a positive value, the <see cref="MultiClassClassifierMetrics.TopKAccuracy"/> will be filled with
+        /// <param name="topK">If given a positive value, the <see cref="MultiClassClassificationMetrics.TopKAccuracy"/> will be filled with
```
Comment (wschin) [inline src/Microsoft.ML.Data/TrainCatalog.cs]: >topK [](start = 25, length = 4)

Not sure if it's topK or just K.

File: src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs (hunk 1 above)
Comment (wschin) [reply src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs]: Agree with Senja. A reference from wiki: [https://en.wikipedia.org/wiki/Multiclass_classification](https://en.wikipedia.org/wiki/Multiclass_classification) 

---
In reply to: [264526176](https://github.com/dotnet/machinelearning/pull/2919#discussion_r264526176) [](ancestors = 264526176)

File: src/Microsoft.ML.Ensemble/Selector/SubModelSelector/BaseSubModelSelector.cs [hunk 5]
```diff
@@ -64,7 +64,7 @@ private IEvaluator GetEvaluator(IHostEnvironment env)
                 case PredictionKind.Regression:
                     return new RegressionEvaluator(env, new RegressionEvaluator.Arguments());
                 case PredictionKind.MultiClassClassification:
```
Comment (wschin) [inline src/Microsoft.ML.Ensemble/Selector/SubModelSelector/BaseSubModelSelector.cs]: >MultiClassClassification [](start = 36, length = 24)

Is changing `MultiClass` to `Multiclass` a part of this PR? If not, please ignore this comment.

**Comment by artidoro** [comment]
Thank you I realized that I forgot to look at the casing `MultiClass` should be replaced by `Multiclass`.

**Comment by artidoro** [comment]
For commit number 4 I have used the following regular expression to replace `MultiClass` with `Multiclass`:

- find: `(?!'\w*)(?!\"\w*)MultiClass(?!\w*')(?!\w*\")`
- replace: `Multiclass`

And I have had to correct for the XML references which ended up not being matched, so I also used the following one:

- find: `(<.*)MultiClass(.*>)`
- replace: `$1Multiclass$2`

File: src/Microsoft.ML.Core/Data/AnnotationUtils.cs [hunk 6]
```diff
@@ -87,7 +87,7 @@ public static class Const
             public static class ScoreColumnKind
             {
                 public const string BinaryClassification = "BinaryClassification";
-                public const string MultiClassClassification = "MultiClassClassification";
+                public const string MulticlassClassification = "MultiClassClassification";
```
Comment (artidoro) [inline src/Microsoft.ML.Core/Data/AnnotationUtils.cs]: >MultiClassClassification [](start = 64, length = 24)

Multiclass

File: src/Microsoft.ML.Core/Prediction/ITrainer.cs [hunk 7]
```diff
@@ -19,7 +19,7 @@ namespace Microsoft.ML
     [BestFriend]
     internal delegate void SignatureBinaryClassifierTrainer();
     [BestFriend]
-    internal delegate void SignatureMultiClassClassifierTrainer();
+    internal delegate void SignatureMulticlassClassificationTrainer();
```
Comment (artidoro) [inline src/Microsoft.ML.Core/Prediction/ITrainer.cs]: >SignatureMulticlassClassificationTrainer [](start = 27, length = 40)

Keep classifier.

File: src/Microsoft.ML.Data/TrainCatalog.cs (hunk 4 above)
Comment (artidoro) [reply src/Microsoft.ML.Data/TrainCatalog.cs]: Yes I think this is k, but I almost prefer it like this.  Just one letter seems worst.

---
In reply to: [264771658](https://github.com/dotnet/machinelearning/pull/2919#discussion_r264771658) [](ancestors = 264771658)

**Comment by artidoro** [comment]
I have updated the PR making sure that `Multiclass` is consistent everywhere in the codebase, and have replaced all occurrences of `MultiClass`. If you could take another look it would be great!

File: src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs [hunk 8]
```diff
@@ -567,7 +567,7 @@ private static VersionInfo GetVersionInfo()
         private readonly ReadOnlyMemory<char>[] _classNames;
         private readonly DataViewType[] _types;
 
-        public MultiClassPerInstanceEvaluator(IHostEnvironment env, DataViewSchema schema, DataViewSchema.Column scoreColumn, string labelCol)
+        public MulticlassPerInstanceEvaluator(IHostEnvironment env, DataViewSchema schema, DataViewSchema.Column scoreColumn, string labelCol)
```
Comment (wschin) [inline src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs]: PerInstance? PerExample? Looks like we don't have a universal name for this.

File: test/Microsoft.ML.Predictor.Tests/ResultProcessor/TestData/MultiClassifier/0.out.txt [hunk 9]
```diff
@@ -1,4 +1,4 @@
-maml.exe TrainTest test=F:\data\MNIST\Train-28x28.txt tr=MultiClassLogisticRegression{l1=0.1 initwts=0.5} loader=TextLoader{col=Label:R4:0 col=Features:R4:1-784} data=F:\data\MNIST\Test-28x28.txt
+maml.exe TrainTest test=F:\data\MNIST\Train-28x28.txt tr=MulticlassLogisticRegression{l1=0.1 initwts=0.5} loader=TextLoader{col=Label:R4:0 col=Features:R4:1-784} data=F:\data\MNIST\Test-28x28.txt
```
Comment (wschin) [inline test/Microsoft.ML.Predictor.Tests/ResultProcessor/TestData/MultiClassifier/0.out.txt]: Command-line breaker... But I feel it's fine.

Comment (Ivanidzo4ka) [reply test/Microsoft.ML.Predictor.Tests/ResultProcessor/TestData/MultiClassifier/0.out.txt]: I think command line is case insensitive, but I can be wrong

---
In reply to: [265270749](https://github.com/dotnet/machinelearning/pull/2919#discussion_r265270749) [](ancestors = 265270749)

**Pull request merged**